src/plugins/discover/public/application/angular/discover.js

```diff
@@ -296,6 +296,14 @@ function discoverController($element, $route, $scope, $timeout, Promise, uiCapab
       appStateContainer,
       data,
       getSavedSearchId: () => savedSearch.id,
+      getSessionName: () => {
+        return (
+          (savedSearch.id && savedSearch.title) ||
```

Comment:
I would suggest that the format for the default session name would be that + a nicely formatted timestamp. Something like `My Saved Search 20210102_204530` (or whatever different format,  that is **sortable**, in case you have multiple items with a similar name).

Reply:
And can we do the same for dashboard (in a separate PR)?

Reply:
Do we need timestamp in the name considering that we have "created_at" in that object + it is there next to the name on mgmt screen? I think created_at should be enough.